### PR TITLE
Implementation for "Manual Set Questlevel range"

### PR DESCRIPTION
**Pull request by MatKier**
Issue: #1732

I added a new setting in the "general" tab under "Quest level options" which changes the level offset to absolute values instead of relative ones.
The slider text and description changes when enabled and the minLevel/maxLevel calculation in Modules\Quest\QuestieQuest.lua uses these absolute values when the setting is enabled instead of calculating the relative ones.
Only downside is see is that AceConfig range widget does not take a function for calculating the min and max values, so I had to set them to stay static values from 0 - 60 (the maxLevelFilter one was already anyway). Users can potentially set absolute ranges that make no sense (e.g.: min:50 max:40), which just shows no quests. But maybe there is a better way to do this?

The localization is only complete from deDE and enUS.

Here are some screenshots showing an absolute setting of 0 to 13:
Absolute level range off

Absolute level range on from 0 to 13

Showing quests from 1 to 13 on the map

### CODE REVIEW

**Comment by Ischtaratu** [comment]
You are my Hero of the Day xD

I looked at your mod and tested it with different chars in different constellations. And it works exactly as I imagined. Thanks alot. If that were to be transferred to the main branch now, it would be really great.

File: Locale/deDE/config.lua [hunk 1]
```diff
@@ -17,6 +17,12 @@ QuestieLocale.locale['deDE'] = {
     ['LOWLEVEL_BELOW_DESC'] = "Quests bis X Level unter deinem eigenen anzeigen. ( Standard: %s )",
     ['LOWLEVEL_ABOVE'] = "Zeige über Level >",
     ['LOWLEVEL_ABOVE_DESC'] = "Quests bis X Level über deinem eigenen anzeigen. ( Standard: %s )",
+    ['ENABLE_MANUAL_OFFSET_ABSOLUTE'] = "Absolute Levelbegrenzen aktivieren",
```
Comment (BreakBB) [inline Locale/deDE/config.lua]:
```suggestion
    ['ENABLE_MANUAL_OFFSET_ABSOLUTE'] = "Absolute Levelbegrenzung aktivieren",
```

File: Locale/deDE/config.lua [hunk 2]
```diff
@@ -17,6 +17,12 @@ QuestieLocale.locale['deDE'] = {
...
+    ['ENABLE_MANUAL_OFFSET_ABSOLUTE_DESC'] = "Ändert die relative Levelbegrenzung zu einem absoluten Wert.",
```
Comment (BreakBB) [inline Locale/deDE/config.lua]:
```suggestion
    ['ENABLE_MANUAL_OFFSET_ABSOLUTE_DESC'] = "Ändert die relative Levelbegrenzung zu einem absoluten Start- und Endwert.",
```

File: Locale/deDE/config.lua [hunk 3]
```diff
@@ -17,6 +17,12 @@ QuestieLocale.locale['deDE'] = {
...
+    ['LEVEL_FROM'] = "Level ab",
+    ['LEVEL_FROM_DESC'] = "Zeige Quests ab level.",
```
Comment (BreakBB) [inline Locale/deDE/config.lua]:
```suggestion
    ['LEVEL_FROM_DESC'] = "Zeige Quests ab dem eingestellten Level.",
```

File: Locale/deDE/config.lua [hunk 4]
```diff
@@ -17,6 +17,12 @@ QuestieLocale.locale['deDE'] = {
...
+    ['LEVEL_TO'] = "Level bis",
+    ['LEVEL_TO_DESC'] = "Zeige Quests bis level.",
```
Comment (BreakBB) [inline Locale/deDE/config.lua]:
```suggestion
    ['LEVEL_TO_DESC'] = "Zeige Quests bis zu dem eingestellten Level.",
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 5]
```diff
@@ -350,14 +350,40 @@ function QuestieOptions.tabs.general:Initialize()
...
+                order = 2.4,
+                name = function() return QuestieLocale:GetUIString('ENABLE_MANUAL_OFFSET_ABSOLUTE'); end,
+                desc = function() return QuestieLocale:GetUIString('ENABLE_MANUAL_OFFSET_ABSOLUTE_DESC'); end,
+                width = 200,
+                disabled = function() return Questie.db.char.lowlevel; end,
+                get = function () return Questie.db.char.manualMinLevelOffsetAbsolute; end,
+                set = function (info, value)
+                    Questie.db.char.manualMinLevelOffsetAbsolute = value
+                    QuestieOptions.AvailableQuestRedraw();
+                    Questie:debug(DEBUG_DEVELOP, QuestieLocale:GetUIString('ENABLE_MANUAL_OFFSET_ABSOLUTE'), value)
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    Questie:Debug(DEBUG_DEVELOP, QuestieLocale:GetUIString('ENABLE_MANUAL_OFFSET_ABSOLUTE'), value)
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 6]
```diff
@@ -350,14 +350,40 @@ function QuestieOptions.tabs.general:Initialize()
...
-                order = 2.4,
-                name = function() return QuestieLocale:GetUIString('LOWLEVEL_BELOW'); end,
-                desc = function() return QuestieLocale:GetUIString('LOWLEVEL_BELOW_DESC', optionsDefaults.global.minLevelFilter); end,
+                order = 2.5,
+                name = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then 
+                        return QuestieLocale:GetUIString('LEVEL_FROM');
+                    else
+                        return QuestieLocale:GetUIString('LOWLEVEL_BELOW'); 
+                        end
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    end
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 7]
```diff
@@ -350,14 +350,40 @@ function QuestieOptions.tabs.general:Initialize()
...
+                end,
+                desc = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then
+                        return QuestieLocale:GetUIString('LEVEL_FROM_DESC');
+                    else
+                        return QuestieLocale:GetUIString('LOWLEVEL_BELOW_DESC', optionsDefaults.global.minLevelFilter); 
+                        end
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    end
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 8]
```diff
@@ -350,14 +350,40 @@ function QuestieOptions.tabs.general:Initialize()
...
+                end,
+            },
             minLevelFilter = {
                 type = "range",
-                order = 2.4,
-                name = function() return QuestieLocale:GetUIString('LOWLEVEL_BELOW'); end,
-                desc = function() return QuestieLocale:GetUIString('LOWLEVEL_BELOW_DESC', optionsDefaults.global.minLevelFilter); end,
+                order = 2.5,
+                name = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    if Questie.db.char.manualMinLevelOffsetAbsolute then 
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 9]
```diff
@@ -350,14 +350,40 @@ function QuestieOptions.tabs.general:Initialize()
...
+                order = 2.5,
+                name = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then 
+                        return QuestieLocale:GetUIString('LEVEL_FROM');
+                    else
+                        return QuestieLocale:GetUIString('LOWLEVEL_BELOW'); 
+                        end
+                end,
+                desc = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    if Questie.db.char.manualMinLevelOffsetAbsolute then
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 10]
```diff
@@ -374,14 +400,26 @@ function QuestieOptions.tabs.general:Initialize()
             },
             maxLevelFilter = {
                 type = "range",
-                order = 2.5,
-                name = function() return QuestieLocale:GetUIString('LOWLEVEL_ABOVE'); end,
-                desc = function() return QuestieLocale:GetUIString('LOWLEVEL_ABOVE_DESC', optionsDefaults.global.maxLevelFilter); end,
+                order = 2.6,
+                name = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    if Questie.db.char.manualMinLevelOffsetAbsolute then
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 11]
```diff
@@ -374,14 +400,26 @@ function QuestieOptions.tabs.general:Initialize()
...
+                        return QuestieLocale:GetUIString('LEVEL_TO');
+                    else
+                        return QuestieLocale:GetUIString('LOWLEVEL_ABOVE'); 
+                        end
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    end
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 12]
```diff
@@ -374,14 +400,26 @@ function QuestieOptions.tabs.general:Initialize()
...
+                end,
+                desc = function()
+                    if(Questie.db.char.manualMinLevelOffsetAbsolute) then
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                    if Questie.db.char.manualMinLevelOffsetAbsolute then
```

File: Modules/Options/GeneralTab/QuestieOptionsGeneral.lua [hunk 13]
```diff
@@ -374,14 +400,26 @@ function QuestieOptions.tabs.general:Initialize()
...
+                    else
+                        return QuestieLocale:GetUIString('LOWLEVEL_ABOVE_DESC', optionsDefaults.global.maxLevelFilter); 
+                    end
+                end,
                 width = "normal",
                 min = 0,
                 max = 60,
                 step = 1,
-                disabled = function() return QuestiePlayer:GetPlayerLevel() == 60; end,
+                disabled = function() return QuestiePlayer:GetPlayerLevel() == 60 and not Questie.db.char.manualMinLevelOffsetAbsolute; end,
```
Comment (BreakBB) [inline Modules/Options/GeneralTab/QuestieOptionsGeneral.lua]:
```suggestion
                disabled = function() return QuestiePlayer:GetPlayerLevel() == 60 and (not Questie.db.char.manualMinLevelOffsetAbsolute); end,
```

File: Modules/Quest/QuestieQuest.lua [hunk 14]
```diff
@@ -1464,6 +1464,10 @@ function QuestieQuest:CalculateAvailableQuests()
     local playerLevel = QuestiePlayer:GetPlayerLevel()
     local minLevel = playerLevel - Questie.db.global.minLevelFilter
     local maxLevel = playerLevel + Questie.db.global.maxLevelFilter
+    if(Questie.db.char.manualMinLevelOffsetAbsolute) then
```
Comment (BreakBB) [inline Modules/Quest/QuestieQuest.lua]:
```suggestion
    if Questie.db.char.manualMinLevelOffsetAbsolute then
```

**Comment by MatKier** [comment]
I'm glad I could help!
Thank you for the positive feedback, I applied all suggested changes.

**Pull request merged**